deploy-to-aks/action.yml

```diff
@@ -0,0 +1,148 @@
+name: Deploy to AKS environment
+description: Deploy a docker image to an AKS environment
+
+inputs:
+  azure-credentials:
+    required: true
+  gcp-wip:
+    required: false
+  environment:
+    description: Environment to deploy to
+    required: true
+  github-token:
+    description: Default Github token retrieved via secrets.GITHUB_TOKEN or PAT with permission to the repository (Required)
+    required: true
+  pr-number:
+    description: PR number for the review app
+    required: false
+  sha:
+    description: Commit sha to be deployed
+    required: true
+  slack-webhook:
+    required: false
+  db-seed:
+    description: run seed command after a review deployment
+    type: boolean
+    required: false
+    default: false
+  smoke-test:
+    description: Enable smoke test after deployment
+    type: boolean
+    required: false
+    default: false
+  healthcheck:
+    description: Health check path
+    required: false
+  terraform-base:
+    description: path to the terraform files
+    required: false
+    default: 'terraform/application'
+
+outputs:
+  environment_url:
+    description: The first external URL for the deployed environment
+    value: ${{ steps.set_outputs.output.DEPLOY_URL }}
+
+runs:
+  using: composite
+  steps:
+    - name: Set Environment variables
+      id: set_env_var
+      shell: bash
+      run: |
+        terraform_version=$(awk '/{/{f=/^terraform/;next}f' ${{ inputs.terraform-base }}/terraform.tf | grep -o [0-9\.]*)
+        echo "TERRAFORM_VERSION=$terraform_version" >> $GITHUB_ENV
+
+    - name: Use Terraform ${{ env.TERRAFORM_VERSION }}
+      uses: hashicorp/setup-terraform@v3
+      with:
+        terraform_version: ${{ env.TERRAFORM_VERSION }}
+        terraform_wrapper: false
+
+    - uses: azure/login@v2
+      if: ${{ ( inputs.db-seed == 'true' && inputs.environment == 'review' ) }}
```

Comment:
Can we make it the responsibility of the caller so we don't hardcode the environment name and allow more flexibility
e.g. `seed-review-app: ${{ inputs.environment == review }}`

Reply:
If you mean to remove the ${{ inputs.environment == review }} I would just be concerned someone could accidentally seed a non-review app, even production.

Reply:
removed